nixos/modules/tasks/filesystems.nix

```diff
@@ -280,7 +280,8 @@ in
     environment.etc.fstab.text =
       let
         fsToSkipCheck = [ "none" "bindfs" "btrfs" "zfs" "tmpfs" "nfs" "vboxsf" "glusterfs" "apfs" ];
-        skipCheck = fs: fs.noCheck || fs.device == "none" || builtins.elem fs.fsType fsToSkipCheck;
+        isBindMount = fs: any (o: o == "bind") fs.options;
+        skipCheck = fs: fs.noCheck || fs.device == "none" || builtins.elem fs.fsType fsToSkipCheck || isBindMount fs;
```

Comment:
it's sort of out of scope of this PR, but we might also want to do this for network filesystems, like cifs or nfs

Reply:
We can just add them to the `fsToSkipCheck` list, right? I was planning to add 9p to the list anyways, I could just add a few more if you like.

Reply:
I'll do that in a different PR though.